modules/fluid_properties/src/userobjects/SalineFluidProperties.C

```diff
@@ -0,0 +1,168 @@
+// std::vector<Real>
+// splitString_float(std::string str)
+//{
+//   std::vector<std::string> list = splitString_str(str);
+//   std::vector<Real> res;
+//   for (std::string s : list)
+//     res.push_back(stof(s));
+//   return res;
+// }
+
+SalineFluidProperties::SalineFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters)
+{
+
+#ifdef SALINE_ENABLED
+
+  std::string propDef = getParam<std::string>("prop_def");
+  _d.load(propDef);
+  bool success = _tp.initialize(&_d);
+  if (!success)
+    mooseError("The initialize of the Saline interface has failed");
+  std::string name = getParam<std::string>("comp_name");
+  std::string comp = getParam<std::string>("comp_val");
+  std::vector<std::string> nameList;
+  MooseUtils::tokenize<std::string>(name, nameList, 1, "-");
+  std::vector<Real> valList;
+  MooseUtils::tokenizeAndConvert<Real>(comp, valList, "-");
+  Real mole_sum = 0.0;
+  for (Real val : valList)
+    mole_sum = mole_sum + val;
+  if (std::abs(mole_sum - 1.0) > 1e-6)
+    mooseError("Mole fractions of defined salt compound do not sum to 1.0.");
+  success = _tp.setComposition(nameList, valList);
+  if (!success)
+    mooseError("The composition set has failed");
+  _fluid_name = name;
+
+#endif
+#ifndef SALINE_ENABLED
+
+  mooseError("Saline was not made available during the build and can not be used.");
+
+#endif
+}
+
+// Unit conversion constants for communicating with Saline
+const Real kPa_to_Pa = 1.0e3;
+const Real Pa_to_kPa = 1.0 / kPa_to_Pa;
+const Real kg_to_g = 1.0e3;
+const Real g_to_kg = 1.0 / kg_to_g;
+const Real m_to_cm = 1.0e2;
+const Real cm_to_m = 1.0 / m_to_cm;
+const Real N_to_mN = 1.0e3;
+const Real mN_to_N = 1 / N_to_mN;
+
+#ifdef SALINE_ENABLED
+
+std::string
+SalineFluidProperties::fluidName() const
+{
+  return _fluid_name;
+}
+
+Real
+SalineFluidProperties::rho_from_p_T(Real pressure, Real temperature) const
+{
+  return _tp.rho_kgm3(temperature, pressure * Pa_to_kPa);
+}
+
+void
+SalineFluidProperties::rho_from_p_T(
+    Real pressure, Real temperature, Real & rho, Real & drho_dp, Real & drho_dT) const
+{
+  rho = rho_from_p_T(pressure, temperature);
+  // Estimate derivatives for now because they are not provided by Saline
+  drho_dp = 0.;
+  drho_dT = (rho_from_p_T(pressure, temperature + 1.0) - rho) / 1.0;
+}
+
+Real
+SalineFluidProperties::cp_from_p_T(Real pressure, Real temperature) const
+{
+  return _tp.cp_kg(temperature, pressure * Pa_to_kPa); // J/kg/K
+}
+
+void
+SalineFluidProperties::cp_from_p_T(
+    Real pressure, Real temperature, Real & cp, Real & dcp_dp, Real & dcp_dT) const
+{
+  cp = cp_from_p_T(pressure, temperature);
+  dcp_dp = 0;
+  dcp_dT = (cp_from_p_T(pressure, temperature + 1.0) - cp) / 1.0;
+}
+
+// Real
+// SalineFluidProperties::enthalpy_from_T(Real temperature) const
+//{
+//   return _tp.h_t_kg(temperature);
+// }
+//
+// Real
+// SalineFluidProperties::T_from_enthalpy(Real enthalpy) const
+//{
+//   return _tp.t_h_kg(enthalpy);
+// }
```

Comment:
it would be h_from_p_T

Reply:
Do you mean this is a function I should be overriding?  I do not see one named `h_from_p_T` in the `SinglePhaseFluidProperties` or `FluidProperties` classes.

Reply:
It s present under a macro in SingleFluidProperties.h
 propfunc(h, p, T)
then for example
```
./FlinakFluidProperties.h:  virtual void h_from_p_T(Real p, Real T, Real & h, Real & dh_dp, Real & dh_dT) const override;
```

Reply:
sounds good. T_from_p_h is a bit more niche btw, I think it s used by Pronghorn - subchannel